custom/icds_reports/utils/aggregation_helpers/distributed/growth_monitoring_forms.py

```diff
@@ -22,7 +22,11 @@ def aggregate(self, cursor):
         agg_query, agg_params = self.aggregation_query()
 
         cursor.execute(
-            f'DELETE FROM "{self.tablename}" WHERE month=%(month)s',
+            f'DELETE FROM "{self.tablename}" WHERE month < %(month)s',
+            {'month': month_formatter(self.month - relativedelta(months=3))}
```

Comment:
maybe a dumb question, but is this expensive if there's no data? In theory it only has to run on the first or something, right?

Reply:
Yeah, because there's an index on month it should not be expensive when there's no data. The expensive part of deletes is usually actually deleting the rows and updating the table definitions and indexes in postgres.